util/http_server.go

```diff
@@ -0,0 +1,47 @@
+package util
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"os"
+)
+
+func GoServeHTTP(ctx context.Context, logger *Logger, addr string, serveMux *http.ServeMux) {
+	s := &http.Server{
+		BaseContext: func(net.Listener) context.Context { return ctx },
+		Addr:        addr,
+		Handler:     serveMux,
+	}
+	lc := net.ListenConfig{}
+	l, err := lc.Listen(ctx, "tcp", s.Addr)
+	if err != nil {
+		logger.PrintError("Error starting HTTP server on %s: %v\n", addr, err)
```

Comment:
One thing it's slightly missing from the previous version is, this error message is too generic and sometimes it'll be hard to tell which part didn't go well. I like that we have this util func so I think it's an acceptable tradeoff, also I can't come up with a better alternative without making things messy, but just figured to mention.